tests/Unit/Listeners/StateHistoryManagerTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Tests\Unit\Listeners;
+
+use Tests\TestCase;
+use App\User;
+use App\AdminUser;
+use App\CentreUser;
+use App\Voucher;
+use Auth;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+
+class StateHistoryManagerTest extends TestCase
+{
+
+    use DatabaseMigrations;
+
+    protected $adminUser;
+    protected $centreUser;
+    protected $user;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->adminUser = factory(AdminUser::class)->create();
+        $this->centreUser = factory(CentreUser::class)->create();
+        $this->user = factory(User::class)->create();
+    }
+
+    /** @test */
+    public function testStateIsUpdatedAsExpected()
+    {
+        Auth::login($this->adminUser);
+        // create a voucher as printed
+
+        $v = factory(Voucher::class, 'requested')->create();
+        $v->applyTransition('order');
+        $v->applyTransition('print');
+        $v->applyTransition('dispatch');
+
+        $state = $v->history()->get(null)->last();
```

Comment:
What doe s arg null do for `get()`?

Does `$v->history->last()` not return what you need?